File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, Integer> myMap = new HashMap<>();
        List<List<String>> result = new ArrayList<>();

        for(int i = 0; i < strs.length; i++){
            String s = strs[i];
            char [] chars = s.toCharArray();
            Arrays.sort(chars);
            s = new String(chars);

            if(!myMap.containsKey(s)){
                myMap.put(s, result.size());
                result.add(new ArrayList<>());
            }
            result.get(myMap.get(s)).add(strs[i]);
        }
        return result;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encodedS = []
        for s in strs:
            encodedS.append(str(len(s)))
            encodedS.append("#")
            encodedS.append(s)
        string = "".join(encodedS)
        return string

    def decode(self, s: str) -> List[str]:
        i = 0
        result = []
        while(i < len(s)):
            j = i
            while(s[j] != "#"):
                j = j + 1
            length = int(s[i:j])
            i = j + 1
            j = i + length
            result.append(s[i:j])
            i = j
        return result
